### ISSUE TITLE
keydispatcher needs major refactor (again)

### ISSUE THREAD

**Issue by eeeeaaii** [eeeeaaii/vodka#68]
Here is the logic flow for keydispatcher:
1. find the context of the selected nex
2. if the input is a naked shift, and we are in the editor, close the editor
3. if there is an open editor, route key to it
4. if key has the same meaning everywhere, handle (meta-c, meta-v, escape), handle it here
5. run "default handle" (which is responsible for returning false if this should be handled by one of the function tables
6. run function from override table, if applicable
7. run function from regular table, if applicable
8. run function from generic table, if applicable
9. input not matched, exit

This is kind of a mess, and we need to perform the following refactors:
1. eliminate override table and context (steps 6 and 1), this is no longer needed
2. either:
    - move default handle to end (after step 8), to be run if input isn't matched by anything else
    - put default handle in the regular table, but with a regex, and use regex matching to find the right table entry
3. move meta-c stuff into generic table, since it shouldn't match anywhere else anyway, and remove step 4
4. move #2 (close editor for naked shift) into the editor itself
5. change to a token/value system, so we can push regex matching up to the beginning (current method is getEventName) rather than doing it in every single defaultHandle method (this actually might mean we could completely delete the defaultHandle methods)

**Comment by eeeeaaii** [comment]
as per https://github.com/eeeeaaii/vodka/issues/77 we cannot remove context and override table. However the way we are doing it now (by passing context type into the default handle function) isn't a great way to do it.

**Comment by eeeeaaii** [comment]
most of this is now complete, with the exception of being able to put regexes in the function table.
